src/sentry/workflow_engine/migration_helpers/rule.py

```diff
@@ -110,3 +115,98 @@ def create_if_dcg(
 def create_workflow_actions(if_dcg: DataConditionGroup, actions: list[dict[str, Any]]):
     # TODO: create actions, need registry
     pass
+
+
+class UpdatedIssueAlertData(TypedDict):
+    name: str
+    conditions: list[dict[str, Any]]
+    action_match: str
+    filter_match: str | None
+    actions: list[dict[str, Any]]
+    environment: int | None
+    owner: Actor | None
+    frequency: int | None
+
+
+def update_migrated_issue_alert(
+    rule: Rule, data: UpdatedIssueAlertData, user: RpcUser | None = None
+):
+    try:
+        alert_rule_workflow = AlertRuleWorkflow.objects.get(rule=rule)
+    except AlertRuleWorkflow.DoesNotExist:
+        logger.exception("AlertRuleWorkflow does not exist", extra={"rule_id": rule.id})
+        return
+
+    workflow: Workflow = alert_rule_workflow.workflow
+    if not workflow.when_condition_group:
+        logger.error(
+            "Workflow does not have a when_condition_group", extra={"workflow_id": workflow.id}
+        )
+        return
+
+    conditions, filters = split_conditions_and_filters(data["conditions"])
+
+    update_dcg(
+        dcg=workflow.when_condition_group,
+        conditions=conditions,
+        match=data["action_match"],
+    )
+
+    try:
+        if_dcg = WorkflowDataConditionGroup.objects.get(workflow=workflow).condition_group
+    except WorkflowDataConditionGroup.DoesNotExist:
+        logger.exception(
+            "WorkflowDataConditionGroup does not exist", extra={"workflow_id": workflow.id}
+        )
+        # if no IF DCG exists, create one
+        if_dcg = create_if_dcg(
+            workflow=workflow, filters=filters, filter_match=data["filter_match"]
+        )
+        WorkflowDataConditionGroup.objects.create(workflow=workflow, condition_group=if_dcg)
+
+    update_dcg(dcg=if_dcg, conditions=filters, match=data["filter_match"])
+
+    delete_workflow_actions(if_dcg=if_dcg, actions=data["actions"], user=user)
```

Comment:
nit: do we need to pass `actions` if we're deleting them?

Reply:
oh you're right, we would just need the `if_dcg` to find the actions right?

Reply:
yes